daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -208,7 +222,152 @@ execCompile inputFile outputFile opts = withProjectRoot $ \relativize -> do
 
 newtype DumpPom = DumpPom{unDumpPom :: Bool}
 
-execPackage :: FilePath -- ^ input file
+execPackageNew :: Int -> Maybe FilePath -> IO ()
+execPackageNew numProcessors mbOutFile =
+    withProjectRoot $ \_relativize -> do
+        project <- readProjectConfig $ ProjectPath "."
+        case parseProjectConfig project of
+            Left err -> throwIO err
+            Right (name, main, exposedModules, version, dependencies) -> do
+                createProjectPackageDb LF.versionDefault dependencies
+                defaultOpts <- Compiler.defaultOptionsIO Nothing
+                let opts =
+                        defaultOpts
+                            { optMbPackageName = Just name
+                            , optThreads = numProcessors
+                            , optWriteInterface = True
+                            }
+                loggerH <- getLogger opts "package"
+                let confFile =
+                        mkConfFile
+                            name
+                            version
+                            LF.versionDefault
+                            exposedModules
+                            dependencies
+                Managed.with (Compiler.newIdeState opts Nothing loggerH) $ \compilerH -> do
+                    darOrErr <-
+                        runExceptT $
+                        Compiler.buildDar
+                            compilerH
+                            main
+                            name
+                            [confFile]
+                            (UseDalf False)
+                    case darOrErr of
+                        Left errs ->
+                            ioError $
+                            userError $
+                            unlines
+                                [ "Creation of DAR file failed:"
+                                , T.unpack $
+                                  Pretty.renderColored $
+                                  Pretty.vcat $
+                                  map prettyDiagnostic $
+                                  Set.toList $ Set.fromList errs
```

Comment:
Do we expect duplicated errors here?

Reply:
Also from the old packaging command, but I prefer to drop the dedup here as I don't understand why we need it.